SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -859,8 +859,9 @@ bool USpatialActorChannel::ReplicateSubobject(UObject* Object, const FReplicatio
 	SCOPE_CYCLE_UOBJECT(ReplicateSubobjectSpecificClass, Object);
 
 	bool bCreatedReplicator = false;
+	bool bOutPreviouslyDormant = false;
 
-	FObjectReplicator& Replicator = FindOrCreateReplicator(Object, &bCreatedReplicator).Get();
+	FObjectReplicator& Replicator = FindOrCreateReplicator(Object, &bCreatedReplicator, &bOutPreviouslyDormant).Get();
```

Comment:
DormantReplicatorMap is publicly available, we should only make engine changes where necessary, I'd rather we rework this so we don't need to change the engine API.

Reply:
```suggestion
	// Set here because FindOrCreateReplicator will remove the replicator from the dormancy map
	const bool bPreviouslyDormant = Connection->DormantReplicatorMap.Find(Object);
	FObjectReplicator& Replicator = FindOrCreateReplicator(Object, &bCreatedReplicator).Get();
```

Reply:
Something like this? I don't love how order dependent this is and reliant on opaque implementation details, but I can see how avoiding engine changes might also have advantages.

Reply:
Ok I am coming around to this @m-samiec are you happy with the change outlined in my suggestion?

Reply:
Lovely